src/connector/src/sink/clickhouse.rs

```diff
@@ -493,8 +493,14 @@ impl Sink for ClickHouseSink {
     type Coordinator = DummySinkCommitCoordinator;
     type LogSinker = DecoupleCheckpointLogSinkerOf<ClickHouseSinkWriter>;
 
+    const SINK_ALTER_CONFIG_LIST: &'static [&'static str] = &["commit_checkpoint_interval"];
```

Comment:
specifying a whitelist for each connector is troublesome and not necessary.
How about utilizing the existing props validation step when creating sink? If it successes, apply the change, reject it  otherwise.

Reply:
May not be a good idea. If we just treat it as recreating the sink, we will not be able to validate the comparison between the existing config and the new config. For example, for kafka sink, it's unlikely for us to change the target topic.

Rust's trait associated const support default value (in this PR we set as &[]), so we don't have to set the whitelist explicitly for all sinks. Only those supporting alter config should set this whitelist.

Reply:
> we will not be able to validate the comparison between the existing config and the new config

why do we need do the comparison?

> For example, for kafka sink, it's unlikely for us to change the target topic.

Indeed, but I cannot find a reason to reject the user doing such change.
I've met request that they want to switch the existing pipeline the prod, then both kafka broker addr and topic name are to be changed. 

---

My point is that it makes no sense to limit what users want to change the settings, as long as they keep the connector type the same.

Reply:
Then how about supporting altering params in this section https://docs.risingwave.com/integrations/destinations/apache-kafka#additional-kafka-parameters as the target for this pr.